packages/cli/src/tools/releaseChecker/getLatestRelease.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * @flow
+ */
+import https from 'https';
+import semver from 'semver';
+import logger from '../logger';
+import cacheManager from './releaseCacheManager';
+
+export type Release = {
+  version: string,
+  changelogUrl: string,
+};
+
+/**
+ * Checks via GitHub API if there is a newer stable React Native release and,
+ * if it exists, returns the release data.
+ *
+ * If the latest release is not newer or if it's a prerelease, the function
+ * will return undefined.
+ */
+export default (async function(currentVersion: string) {
+  logger.debug('Checking for a newer version of React Native');
+  try {
+    logger.debug(`Current version: ${currentVersion}`);
+
+    const cachedLatest: string = cacheManager.get('latestVersion');
+
+    if (cachedLatest) {
+      logger.debug(`Cached release version: ${cachedLatest}`);
+    }
+
+    logger.debug('Checking for newer releases on GitHub');
+    const eTag = cacheManager.get('eTag');
+    const latestVersion = await getLatestRnDiffPurgeVersion(eTag);
+    logger.debug(`Latest release: ${latestVersion}`);
+
+    if (
+      semver.compare(latestVersion, currentVersion) === 1 &&
+      !semver.prerelease(latestVersion)
+    ) {
+      return {
+        version: latestVersion,
+        changelogUrl: buildChangelogUrl(latestVersion),
+      };
+    }
+  } catch (e) {
+    logger.debug(
+      'Something went wrong with remote version checking, moving on',
+    );
```

Comment:
Print the error in debug
```suggestion
    );
    logger.debug(e);
```